### PR TITLE
fix: fix `useActionData`/`useLoaderData` usage

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
See https://github.com/remix-run/remix/releases/tag/remix@1.6.5

### CODE REVIEW

**Comment by machour** [comment]
@MichaelDeBoey this is a big PR and makes it a little hard to validate.. I see you didn't use `LoaderArgs` in the few changes I looked at, should you?

**Comment by MichaelDeBoey** [comment]
> this is a big PR and makes it a little hard to validate

@machour I discussed it with @mcansh to have one big PR for all community examples & have separate PRs for each official example.

> I see you didn't use `LoaderArgs` in the few changes I looked at, should you?

I'm using `ActionArgs`/`LoaderArgs` whenever they're _actually_ used.
No need to add the type if it's not used.
This sometimes results in a removal of the type import.

**Comment by machour** [comment]
> I'm using ActionArgs/LoaderArgs whenever they're actually used.

Aren't they needed for a proper inference?

**Comment by MichaelDeBoey** [comment]
@machour No they aren't
Problem with previous usage & inference was the `ActionFunction`/`LoaderFunction` types
Now we have a normal function, where TS will infer the return type

File: firebase/app/routes/login.tsx [hunk 1]
```diff
@@ -31,13 +27,13 @@ export const loader: LoaderFunction = async ({ request }) => {
...
-  return json<LoaderData>({ apiKey, domain }, { headers });
+  return json({ apiKey, domain }, { headers });
 };
 
-interface ActionData {
+type ActionData = {
   error?: string;
-}
-export const action: ActionFunction = async ({ request }) => {
+};
```
[the remark this answers is not in the document]
Comment (MichaelDeBoey) [reply firebase/app/routes/login.tsx:35]: It's used in a `useState`
https://github.com/remix-run/examples/blob/d8ced0e912440d2b28ae4c3b12187bfd4669a449/firebase/app/routes/login.tsx#L72

**Comment by mcansh** [comment]
got through 90 of the easier to visually check ones, running a lil bash script to install, build, and type check the rest 😅 - almost there!

**Comment by mcansh** [comment]
> @mcansh Would be nice if we could add that script to a workflow to check on PRs (as explained in #92)

def, for now it's this mess lol https://gist.github.com/mcansh/ec38261fd0356ba050b9c10f1ed78c61

**Pull request merged**